slang-checks/src/main/java/com/sonarsource/slang/checks/EmptyBlockCheck.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * SonarSource SLang
+ * Copyright (C) 2009-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package com.sonarsource.slang.checks;
+
+import com.sonarsource.slang.api.BlockTree;
+import com.sonarsource.slang.api.MatchTree;
+import com.sonarsource.slang.checks.api.InitContext;
+import com.sonarsource.slang.checks.api.SlangCheck;
+import com.sonarsource.slang.impl.FunctionDeclarationTreeImpl;
+
+public class EmptyBlockCheck implements SlangCheck {
+
+  private static final String MESSAGE = "Either remove or fill this block of code.";
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(BlockTree.class, (ctx, blockTree) -> {
+      if (!(ctx.parent() instanceof FunctionDeclarationTreeImpl)) {
+        boolean noStatementsOrExpressions = blockTree.statementOrExpressions().isEmpty();
+        boolean noComments = blockTree.metaData().commentsInside().isEmpty();
```

Comment:
We can factorize that code in a method ("hasNoComment") and call it only when `noStatementsOrExpressions` is true.

Reply:
Fixed at c8d367e